src/script/view_model/CallingViewModel.ts

```diff
@@ -251,23 +284,22 @@ export class CallingViewModel {
       }
     });
 
+    //update epoch info when AVS requests the list of clients
     this.callingRepository.onRequestClientsCallback(async (conversationId: QualifiedId) => {
-      // TODO update epoch info when AVS requests the list of clients
       const mlsService = core.service?.mls;
       if (!mlsService) {
         throw new Error('mls service was not initialised');
       }
 
-      const subconversation = await core.service?.conversation.getSubconversation(conversationId, 'conference' as any);
-
-      if (!subconversation) {
-      }
+      const subconversation = await mlsService.getConferenceSubconversation(conversationId);
 
-      //const {epoch, keyLength, secretKey, members} = await subscribeToEpochUpdates(conversationId, subconversation);
-
-      //this.callingRepository.setEpochInfo(conversationId, {epoch, keyLength, secretKey}, members);
+      const {epoch, keyLength, secretKey, members} = await getSubconversationEpochInfo(conversationId, subconversation);
+      this.callingRepository.setEpochInfo(conversationId, {epoch, keyLength, secretKey}, members);
     });
 
+    //once we leave the call, we unsubscribe from all the events we've subscribed to during the call
+    this.callingRepository.onLeaveCall(() => this.callSubscriptions.forEach(unsubscribe => unsubscribe()));
```

Comment:
We would probably need to filter the subscription to the call that is actually left. 
We currently do not implement multi ongoing calls, but we should make sure that  this is not going to cause trouble in the future (in case mulit ongoing calls are implemented)

Reply:
Maybe a `Map` would be better suited for this. 

```js
this.callingRepository.onLeaveCall((conversationId) => {
  ongoingSubscription.get(convId)?.();
  ongoingSubscription.delete(convId)
});
```

WDYT?